src/lambda/trans.lisp

```diff
@@ -98,14 +281,12 @@ iterated, so is the uncurrying."
                       (rec context term)))
                (unit
                 (terminal (stlc-ctx-to-mu context)))
-               ((left term)
-                (let ((tott (ttype tterm)))
-                  (comp (->left (mcar tott) (mcadr tott))
-                        (rec context term))))
-               ((right term)
-                (let ((tott (ttype tterm)))
-                  (comp (->right (mcar tott) (mcadr tott))
-                        (rec context term))))
+               ((left rty term)
+                (comp (->left (ttype term) rty)
+                      (rec context term)))
+               ((right lty term)
+                (comp (->right lty (ttype term))
+                      (rec context term)))
```

Comment:
You should do these kinds of changes in a different topic, if they are unrelated to your changes.